src/Illuminate/Hashing/SodiumHasher.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+namespace Illuminate\Hashing;
+
+use Illuminate\Contracts\Hashing\Hasher as HasherContract;
+
+class SodiumHasher implements HasherContract
+{
+    /**
+     * Check if the given hash has been hashed using the given options.
+     *
+     * @param  string  $unusedHashedValue
+     * @param  array   $unusedOptions
+     * @return bool
+     */
+    public function needsRehash($unusedHashedValue, array $unusedOptions = [])
+    {
+        return false;
+    }
+
+    /**
+     * Hash the given value.
+     *
+     * @param  string  $value
+     * @param  array   $unusedOptions
+     * @return string
+     *
+     * @throws \RuntimeException
+     */
+    public function make($value, array $unusedOptions = [])
+    {
+        return \Sodium\crypto_pwhash_scryptsalsa208sha256_str(
```

Comment:
you must always import things - laravel code never uses a leading `\` to get to a namespace